src/Peripherals/OpeningHours.php

```diff
@@ -0,0 +1,144 @@
+
+    public $chat;
+    public $call;
+    public CustomerExperienceSettings $settings;
+    public string $chatButtonAction;
+    public $chatButtonActionProvider = null;
+    protected $chatButtonActionToken;
+
+    public function rules()
+    {
+        return [
+            'chat.*.enabled' => ['required'],
+            'chat.*.opening_time' => ['required'],
+            'chat.*.closing_time' => ['required'],
+            'call.*.enabled' => ['required'],
+            'call.*.opening_time' => ['required'],
+            'call.*.closing_time' => ['required'],
+            'chatButtonActionProvider' => ['required'],
+            'chatButtonAction' => ['required'],
+        ];
+    }
+
+    public function mount(): void
+    {
+        $this->seedDatabase();
+
+        $this->chat = OpeningHoursModel::query()->chat()->get();
+        $this->call = OpeningHoursModel::query()->call()->get();
+        $this->settings = app(CustomerExperienceSettings::class);
+        $this->chatButtonActionProvider = $this->settings->chat_button_action_provider;
+        $this->chatButtonAction = $this->settings->chat_button_action;
+    }
+
+    public function seedDatabase(): void
+    {
+        $defaultOpeningTime = Carbon::createFromTime(hour: 9, tz: config('app.timezone'));
+        $defaultClosingTime = Carbon::createFromTime(hour: 18, tz: config('app.timezone'));
+        $days = [CarbonInterface::MONDAY, CarbonInterface::TUESDAY, CarbonInterface::WEDNESDAY, CarbonInterface::THURSDAY, CarbonInterface::FRIDAY, CarbonInterface::SATURDAY, CarbonInterface::SUNDAY];
+
+        if (OpeningHoursModel::call()->count() == 0) {
+            foreach ($days as $day) {
+                OpeningHoursModel::query()->create([
+                    'day' => $day,
+                    'opening_time' => $defaultOpeningTime->format('H:i'),
+                    'closing_time' => $defaultClosingTime->format('H:i'),
+                    'enabled' => ! in_array($day, [CarbonInterface::SATURDAY, CarbonInterface::SUNDAY]),
+                    'type' => OpeningHoursModel::CALL,
+                ]);
+            }
+        }
+
+        if (OpeningHoursModel::chat()->count() == 0) {
+            foreach ($days as $day) {
+                OpeningHoursModel::query()->create([
+                    'day' => $day,
+                    'opening_time' => $defaultOpeningTime->format('H:i'),
+                    'closing_time' => $defaultClosingTime->format('H:i'),
+                    'enabled' => ! in_array($day, [CarbonInterface::SATURDAY, CarbonInterface::SUNDAY]),
+                    'type' => OpeningHoursModel::CHAT,
+                ]);
+            }
+        }
+    }
+
+    public function chatButtonActionProviders(): Collection
+    {
+        return collect(app(Strata::class)->appTokens())
+            ->put('custom', [
+                AppToken::name('Custom action')
+                    ->group('custom')
+                    ->key('custom')
+                    ->value('')
```

Comment:
@tonning Thank you for extending our PR. This looks fantastic, especially with the ability to provide so many providers compared to the one we wanted to tie to only Zendesk. The only issue I'm getting while QA'ing this locally is the error of [Undefined array key "" ](https://flareapp.io/share/q5YJOZVm). This error disappeared when `chat_button_action_provider` payload was set to `custom` instead of `""`.